pallets/xor-fee/src/lib.rs

```diff
@@ -428,6 +429,39 @@ pub trait WithdrawFee<T: Config> {
     ) -> Result<(T::AccountId, Option<NegativeImbalanceOf<T>>), DispatchError>;
 }
 
+/// Trait for dynamic fee update via multiplier
+pub trait CalculateMultiplier<AssetId, Error> {
+    /// Parameters:
+    /// `input_asset` is asset id which price should be fetched;
+    /// `ref_asset` is asset id in which price will be fetched
+    fn fetch_price_to_reference_asset(
+        input_asset: &AssetId,
+        ref_asset: &AssetId,
+    ) -> Result<Balance, Error>;
+
+    /// Parameters:
+    /// `input_asset` is asset id which price should be fetched;
+    /// `ref_asset` is asset id in which price will be fetched
+    fn calculate_multiplier(input_asset: &AssetId, ref_asset: &AssetId)
+        -> Result<FixedU128, Error>;
+}
+
+impl<AssetId> CalculateMultiplier<AssetId, DispatchError> for () {
+    fn fetch_price_to_reference_asset(
+        _input_asset: &AssetId,
+        _ref_asset: &AssetId,
+    ) -> Result<Balance, DispatchError> {
+        Err(DispatchError::CannotLookup)
+    }
+
+    fn calculate_multiplier(
+        _input_asset: &AssetId,
+        _ref_asset: &AssetId,
+    ) -> Result<FixedU128, DispatchError> {
+        Err(DispatchError::CannotLookup)
+    }
+}
```

Comment:
Probably we can move it in `common/traits.rs`

Reply:
We may, but it is used only there and I don't know where it may be used else